src/services/kuzzleWrapper-v1.ts

```diff
@@ -261,91 +261,73 @@ export class KuzzleWrapperV1 {
-          if (mappingAttrs[filter.attribute].type === 'keyword') {
-            formattedFilter.bool.must.push({
-              regexp: {
-                [filter.attribute]:
-                  '.*' + buildCaseInsensitiveRegexp(filter.value) + '.*'
-              }
-            })
-          }
-        } else if (filter.operator === 'not_contains') {
-          if (mappingAttrs[filter.attribute].type === 'text') {
-            formattedFilter.bool.must_not.push({
-              match_phrase_prefix: { [filter.attribute]: filter.value }
-            })
-          }
-          if (mappingAttrs[filter.attribute].type === 'keyword') {
-            formattedFilter.bool.must_not.push({
-              regexp: {
-                [filter.attribute]:
-                  '.*' + buildCaseInsensitiveRegexp(filter.value) + '.*'
+        const mustOperators = ['contains', 'equal', 'range', 'exists']
+        const mustNotOperators = ['not_contains', 'not_equal', 'not_exists']
+
+        let verb: any = null
+        if (mustOperators.includes(filter.operator)) {
+          verb = 'must'
+        } else if (mustNotOperators.includes(filter.operator)) {
+          verb = 'must_not'
+        }
+
+        let filterToAdd: any = null
+        switch (filter.operator) {
+          case 'contains':
+          case 'not_contains':
+            if (filter.attribute === '_id') {
+              filterToAdd = {
+                match: { [filter.attribute]: filter.value }
               }
-            })
-          }
-        } else if (filter.operator === 'equal') {
-          formattedFilter.bool.must.push({
-            range: {
-              [filter.attribute]: {
-                gte: filter.value,
-                lte: filter.value
+            } else if (mappingAttrs[filter.attribute].type === 'text') {
+              filterToAdd = {
+                match_phrase_prefix: { [filter.attribute]: filter.value }
               }
-            }
-          })
-        } else if (filter.operator === 'not_equal') {
-          formattedFilter.bool.must_not.push({
-            range: {
-              [filter.attribute]: {
-                gte: filter.value,
-                lte: filter.value
+            } else if (mappingAttrs[filter.attribute].type === 'keyword') {
+              filterToAdd = {
+                regexp: {
+                  [filter.attribute]:
+                    '.*' + buildCaseInsensitiveRegexp(filter.value) + '.*'
+                }
               }
             }
-          })
-        } else if (filter.operator === 'range') {
-          const range = { range: {} }
-          if (filter.gt_value && filter.lt_value) {
-            range.range = {
-              [filter.attribute]: {
-                gt: filter.gt_value,
-                lt: filter.lt_value
+            break
+          case 'equal':
+          case 'not_equal':
+            filterToAdd = {
+              range: {
+                [filter.attribute]: {
+                  gte: filter.value,
```

Comment:
What if the type of the field is not numeric?

Reply:
I didn't change the logic in that part, I just refactored it using a switch case
you can see [here](https://github.com/kuzzleio/kuzzle-admin-console/blob/cdfd2fed39032afe167412c14110ef89e2a8ce7e/src/services/kuzzleWrapper-v1.ts#L293) that the 3 `else if` of `equal` `not_equal` and `range` where using a ES `range` filter so I let that filter.
If we need to change that, I would prefer doing it in another PR